pigeons/playback_platform_pigeon.dart

```diff
@@ -334,3 +336,7 @@ class MediaItemTransitionEvent implements PlayerEvent {
   MediaItem? mediaItem;
   MediaItemTransitionEvent({required this.playerId, this.mediaItem});
 }
+
+class AnalyticsEvent {
+  late Map<String?, String?>? data;
```

Comment:
There more I think about it, the more i think its probably a better idea to pass the unique id in the mediaitem.
Or if you want to keep this approach, i would have the data more typed:
```
late String playerId;
late AnalyticsEventType type; // an enum

```